### PR TITLE
Make docs/providers/future_provider compatible with toggles

### PR DESCRIPTION
**Pull request by mansourzaki**
docs/providers/future_provider now is comaptible with toggles but I just need to know if the examples are okay or if there is a reference have all the new examples

### CODE REVIEW

**Comment by rrousselGit** [comment]
Hello!
Thanks for your work.

Some links/imports are broken. Could you fix the CI? 

```
12:06:55 PM: Module not found: Error: Can't resolve '/docs/providers/future_provider/config_provider.dart' in '/opt/build/repo/website/i18n/ko/docusaurus-plugin-content-docs/current/providers'
Module not found: Error: Can't resolve '/docs/providers/future_provider/config_consumer.dart' in '/opt/build/repo/website/i18n/ko/docusaurus-plugin-content-docs/current/providers'
Module not found: Error: Can't resolve '/docs/providers/future_provider/config_provider.dart' in '/opt/build/repo/website/i18n/ko/docusaurus-plugin-content-docs/current/providers'
Module not found: Error: Can't resolve '/docs/providers/future_provider/config_consumer.dart' in '/opt/build/repo/website/i18n/ko/docusaurus-plugin-content-docs/current/providers'
```

File: website/docs/providers/future_provider/config_consumer/cgen_config_consumer.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
```
Comment (rrousselGit) [inline website/docs/providers/future_provider/config_consumer/cgen_config_consumer.dart:1]: For the same of easy copy-paste, could could name the files in those folders:

```
codegen.dart
hooks_codegen.dart
raw.dart
hooks.dart
index.tsx
```
?


Reusing the same names allows cloning the folder without having to modify the `index.tsx` everytime.

File: website/docs/providers/future_provider.mdx [hunk 2]
```diff
@@ -36,11 +37,17 @@ Creating the configuration would be done with your typical async/await
...
 Using Flutter's asset system, this would be:
 
-<CodeBlock>{trimSnippet(configProvider)}</CodeBlock>
+<AutoSnippet language="dart" {...configProvider}></AutoSnippet>
+
+<ConditionalSnippet codegen={true}>
+  Finally, run the code-generator with{" "}
+  <code>flutter pub run build_runner watch</code>
+</ConditionalSnippet>
+
```
Comment (rrousselGit) [inline website/docs/providers/future_provider.mdx:46]: I don't think this bit is necessary. It's already mentioned in the getting started doc.

If we mention it in other pages that use codegen too, we'll end-up mentioning it absolutely everywhere

File: website/docs/providers/future_provider/config_provider/cgen_provider.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// ignore_for_file: avoid_unused_constructor_parameters
```
Comment (rrousselGit) [inline website/docs/providers/future_provider/config_provider/cgen_provider.dart:1]: Same comment about file names here too

**Comment by rrousselGit** [comment]
It's looking nice, thanks!

**Comment by rrousselGit** [comment]
Could you change the top of the future_provider docs to:

```md
---
title: FutureProvider
version: 1
```
?

Then you'll have to run `cd website && dart run src/outdated_docs_generator.dart`

File: website/docs/providers/future_provider/config_consumer/cgen_config_consumer.dart (hunk 1 above)
Comment (mansourzaki) [reply website/docs/providers/future_provider/config_consumer/cgen_config_consumer.dart:1]: Oh, I will work on it now

**Comment by mansourzaki** [comment]
> Could you change the top of the future_provider docs to:
> 
> ```lisp
> ---
> title: FutureProvider
> version: 1
> ---
> ```
> 
> ?
> 
> Then you'll have to run `cd website && dart run src/outdated_docs_generator.dart`

Sure, I also will fix all what you've mentioned above and the CI fails because of the korean version uses the same furture_provider.mdx  and I didn't edit the import path for new examples and also I will add the example for all other lanugages

**Comment by rrousselGit** [comment]
Perfect, thanks for the help!

**Pull request merged**